translate/src/context/EntitiesList.tsx

```diff
@@ -0,0 +1,29 @@
+import { createContext, useState } from 'react';
+
+type EntitiesList = Readonly<{
+  visible: boolean;
+  toggleEntitiesList(): void;
+}>;
+
+const initEntitiesList: EntitiesList = {
+  visible: false,
+  toggleEntitiesList: () => {},
+};
+
+export const EntitiesList = createContext(initEntitiesList);
+
+export function EntitiesListProvider({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const [state, setState] = useState(() => ({
+    ...initEntitiesList,
+    toggleEntitiesList: () =>
+      setState((prev) => ({ ...prev, visible: !prev.visible })),
```

Comment:
Looking at how this is used, we don't really want a toggle function, but something like `showEntityList(visible: boolean)` -- in the places we want to set the visibility, we want to explicitly show or hide it.